client/post-editor/editor-more-options/copy-post.jsx

```diff
@@ -0,0 +1,118 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import { getSelectedSite } from 'state/ui/selectors';
+import AccordionSection from 'components/accordion/section';
+import Button from 'components/button';
+import Dialog from 'components/dialog';
+import FormSectionHeading from 'components/forms/form-section-heading';
+import Gridicon from 'components/gridicon';
+import InfoPopover from 'components/info-popover';
+import PostSelector from 'my-sites/post-selector';
+
+class EditorMoreOptionsCopyPost extends Component {
+
+	static propTypes = {
+		site: PropTypes.object.isRequired,
+		translate: PropTypes.func,
+	};
+
+	constructor() {
+		super( ...arguments );
+		this.state = this.resetState();
+	}
+
+	resetState() {
+		return {
+			selectedPostId: null,
+			showDialog: false,
+		};
+	}
+
+	openDialog = event => {
+		event.preventDefault();
+
+		this.setState( {
+			showDialog: true,
+		} );
+	}
+
+	closeDialog = () => {
+		this.setState( this.resetState() );
+	}
+
+	setPostToCopy = post => {
+		this.setState( {
+			selectedPostId: post.ID,
+		} );
+	}
+
+	goToNewDraft = () => {
+		if ( '' !== this.state.selectedPostId ) {
+			page.redirect( `/post/${ this.props.site.slug }?copy=${ this.state.selectedPostId }` );
+			this.closeDialog();
+		}
+	}
+
+	render() {
+		const { translate, site } = this.props;
+		const buttons = [ {
+			action: 'cancel',
+			label: translate( 'Cancel' ),
+		}, {
+			action: 'copy',
+			label: translate( 'Overwrite' ),
+			isPrimary: true,
+			disabled: ! this.state.selectedPostId,
+			onClick: this.goToNewDraft,
+		} ];
+		return (
+			<AccordionSection className="editor-more-options__copy-post">
+				<div className="editor-drawer__label-text">
+					{ translate( 'Copy Post' ) }
+					<InfoPopover position="top left">
+						{ translate( "Pick a post and we'll copy the title, content, tags and categories. " ) }
+					</InfoPopover>
+				</div>
+				<Button borderless compact={ true } onClick={ this.openDialog }>
+					<Gridicon icon="clipboard" /> { translate( 'Select a post to copy' ) }
+				</Button>
+				<Dialog
+					autoFocus={ false }
+					isVisible={ this.state.showDialog }
+					buttons={ buttons }
+					onClose={ this.closeDialog }
+					additionalClassNames="copy-post__select-post-dialog"
+				>
+					<FormSectionHeading>
+						{ translate( 'Select a post to copy' ) }
+					</FormSectionHeading>
+					<p>
+						{ translate( "Pick a post and we'll copy the title, content, tags and categories. " ) }
+					</p>
+					<PostSelector
+						siteId={ site.ID }
```

Comment:
On an un-"primed" Redux state, `site` can be `null` (see [selector docs and implementation](https://github.com/Automattic/wp-calypso/blob/69c39a108f90dcd6a2f1ad35183133f4cb2801af/client/state/ui/selectors.js#L13-L26)). You'll see this to be the case if you run Calypso with state persistence disabled:

```
DISABLE_FEATURES=persist-redux make run
```

In which case, if this component were rendered before the sites were loaded, it would result in an uncaught error.